src/_common/user/action-reasons.ts

```diff
@@ -0,0 +1,71 @@
+import { Translate } from '../translate/translate.service';
+
+export const REASON_SPAM = 'spam';
+export const REASON_OFF_TOPIC = 'off-topic';
+export const REASON_ABUSE = 'abuse';
+export const REASON_OTHER = 'other';
+
+/**
+ * Reason object (`value => display`) of reasons to block a user from a community.
+ */
+export function getCommunityBlockReasons() {
+	return getReasons([REASON_SPAM, REASON_OFF_TOPIC, REASON_ABUSE, REASON_OTHER]);
+}
+
+/**
+ * Reason object (`value => display`) of reasons to eject a post from a community.
+ */
+export function getCommunityEjectPostReasons() {
+	return getReasons([REASON_SPAM, REASON_OFF_TOPIC, REASON_ABUSE, REASON_OTHER]);
+}
+
+/**
+ * Reason object (`value => display`) of reasons to move a post to a different community channel.
+ */
+export function getCommunityMovePostReasons() {
+	return getReasons([REASON_SPAM, REASON_OFF_TOPIC, REASON_OTHER]);
+}
+
+/**
+ * Gets the translatable display text for a predefined reason, or returns the input
+ * custom reason when none of the predefined ones match.
+ */
+export function getSingleReasonText(reason: string) {
+	return getReasons([reason])[reason];
+}
+
+/**
+ * Generates an object map of reason identifier strings to their translated display strings.
+ */
+function getReasons(reasons: string[]) {
+	const reasonsMap = {} as { [reason: string]: string };
+
+	for (const reason of reasons) {
+		let reasonText = null;
+		const reasonTest = reason
+			.toLowerCase()
+			.replace('_', '-')
+			.replace(' ', '-');
```

Comment:
I don't get why this is needed.

Reply:
It's so when someone types "Off Topic" into the input for example, we can show the actual reason for "off-topic". Just wanted to prevent people being stupid and putting one of the predefined reasons as a custom reason, just with an underscore or a space.